Lib/gftools/fix.py

```diff
@@ -605,28 +605,30 @@ def drop_mac_names(ttfont):
     return changed
 
 
-def _blank_glyph_to_gid1(ttfont):
+def _add_blank_glyph_to_gid1(ttfont):
     from nanoemoji.reorder_glyphs import reorder_glyphs
     from nanoemoji.svg import _ensure_ttfont_fully_decompiled
-
-    second_glyph = ttfont.getGlyphOrder()[1]
+    from fontTools.ttLib.tables._g_l_y_f import Glyph
+    glyph_order = ttfont.getGlyphOrder()
     glyf_table = ttfont["glyf"]
-    # Will also work for composite glyphs since these have a -1 contour count
-    if glyf_table[second_glyph].numberOfContours > 0:
-        _ensure_ttfont_fully_decompiled(ttfont)
-        glyph_order = ttfont.getGlyphOrder()
-        empty_glyph = next(
-            (g for g in glyph_order if glyf_table[g].numberOfContours == 0),
-            None
-        )
-        if empty_glyph is None:
-            raise ValueError(
-                "Font contains no empty glyphs. Please include a space or .null glyph"
-            )
-        new_order = glyph_order
-        new_order.remove(empty_glyph)
-        new_order.insert(1, empty_glyph)
-        reorder_glyphs(ttfont, new_order)
+
+    gid1 = glyph_order[1]
+    if glyf_table[gid1].numberOfContours == 0:
+        return
+    _ensure_ttfont_fully_decompiled(ttfont)
+    hmtx = ttfont["hmtx"]
+    blank_glyph = Glyph()
+    blank_name = "blank"
+    glyf_table[blank_name] = blank_glyph
```

Comment:
the problem with using `_g_l_y_f_table.__setitem__` method is that it automatically appends to the TTFont's  glyphOrder list without also invalidating the {glyphNames:glyphID} cache, which means TTFont.getGlyphID could return incorrect result. Instead you can set the new glyphs directly inside the glyf's `glyphs` dict and call TTFont.setGlyphOrder at the end with the new desired order, which automatically triggers a rebuild of glyphID cache.
We do the same in nanoemoji.glue_together module where we add the layer glyphs used by COLR table when we maximum_color converts from OT-SVG => COLR:
https://github.com/googlefonts/nanoemoji/blob/2ffdd23faad153c5cec763b6e210befc9d704cd1/src/nanoemoji/glue_together.py#L53-L72
Also see related fonttools issue: https://github.com/fonttools/fonttools/issues/2605

Reply:
Thanks. I will explore this now.